BoxContentSDK/BoxContentSDK/Requests/BOXFileDeleteRequest.m

```diff
@@ -42,52 +63,109 @@ - (BOXAPIOperation *)createOperation
+    
+    if ([self shouldPerformBackgroundOperation] == YES) {
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:url
+                                                             HTTPMethod:BOXAPIHTTPMethodDELETE
+                                                  queryStringParameters:nil
+                                                         bodyDictionary:nil
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateId];
+        
+        NSString *requestDirectory = self.requestDirectoryPath;
+        NSString *destinationPath = [requestDirectory stringByAppendingPathComponent:self.associateId];
+        dataOperation.destinationPath = destinationPath;
+        
+        return dataOperation;
+    } else {
+        BOXAPIJSONOperation *JSONoperation = [self JSONOperationWithURL:url
+                                                             HTTPMethod:BOXAPIHTTPMethodDELETE
+                                                  queryStringParameters:nil
+                                                         bodyDictionary:nil
+                                                       JSONSuccessBlock:nil
+                                                           failureBlock:nil];
+        
+        if ([self.matchingEtag length] > 0) {
+            [JSONoperation.APIRequest setValue:self.matchingEtag forHTTPHeaderField:BOXAPIHTTPHeaderIfMatch];
+        }
+        
+        return JSONoperation;
     }
-
-    return JSONoperation;
 }
 
 - (void)performRequestWithCompletion:(BOXErrorBlock)completionBlock
 {
-    BOOL isMainThread = [NSThread isMainThread];
-    BOXAPIJSONOperation *folderOperation = (BOXAPIJSONOperation *)self.operation;
-    
-    folderOperation.success = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSDictionary *JSONDictionary) {
-
-        if ([self.cacheClient respondsToSelector:@selector(cacheFileDeleteRequest:error:)]) {
-            [self.cacheClient cacheFileDeleteRequest:self error:nil];
-        }
-        if (completionBlock) {
-            [BOXDispatchHelper callCompletionBlock:^{
-                completionBlock(nil);
-            } onMainThread:isMainThread];
-        }
-    };
-    folderOperation.failure = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error, NSDictionary *JSONDictionary) {
-
-        if ([self.cacheClient respondsToSelector:@selector(cacheFileDeleteRequest:error:)]) {
-            [self.cacheClient cacheFileDeleteRequest:self error:error];
-        }
-        if (completionBlock) {
-            [BOXDispatchHelper callCompletionBlock:^{
-                completionBlock(error);
-            } onMainThread:isMainThread];
-        }
-    };
-    [self performRequest];
+    if ([self shouldPerformBackgroundOperation] == YES) {
```

Comment:
Seems like we could merge the `if` and the `else` blocks, if we just ensure that we're doing the right thing on `destinationPath == nil`.

Reply:
We have different operation types. I think we can leave this as is and migrate all to a common background operation.